src/SFML/Graphics/Shape.cpp

```diff
@@ -129,6 +135,22 @@ float Shape::getOutlineThickness() const
 }
 
 
+////////////////////////////////////////////////////////////
+void Shape::setMiterLimit(float miterLimit)
+{
+    assert(miterLimit >= 1.f && "Shape::setMiterLimit(float) cannot set miter limit to a value lower than 1");
+    m_miterLimit = std::max(1.f, miterLimit);
```

Comment:
Not sure if we should write "defensive" code like that. Yes, it will warn in debug mode, but it will also silently "work" in release mode when you set a value less than 1.

What's the effect of allowing negative numbers? Would it crash or just render distorted?

Reply:
Miter limit lower than 1 is asking outline not to grow bigger around angle than the outline thickness. Basically the limit is always reached. It makes little sense.
Having miter limit lower than 1 gives strange look and can run into Vector2 assert about division by 0.

Here are some example of what it looks like starting at 1 (the lowest valid limit) and going down. The result is especially wrong when angles are large enough (3rd example).

![miter limit](https://github.com/SFML/SFML/assets/1485306/add3d165-60da-4752-9b23-15d007a42f38)

Reply:
@Bromeon Since we've had discussion about bad input handling, what's your view on doing assert + limit?

Reply:
I'm not sure if we do `assert` + limit anywhere else. It may be more in line with other APIs if we assert but otherwise allow the original value to be used. For example `sf::WindowBase::setMinimumSize` asserts a valid minimum size (it can't be large than the maximum window size) but it doesn't clamp the minimum size to be valid. The invalid value continues to be used and whatever happens after that is unspecified.

Reply:
Difficult question. In an ideal world, the developer would catch the assert during development.

But _if_ it slips through, it's probably better if it doesn't cause catastrophic damage.
However, "distorted rendering" is not catastrophic in my opinion (on the contrary, it could _increase_ chances of finding such bugs).

This is a hard problem, and I haven't seen any library/framework/engine handle errors always gracefully. Attempts to do so often result in excessively verbose error handling code, and other approaches (exceptions, panics, aborts) often crash the app/game for minuscule bugs.

Reply:
Ok, I removed the `std::max` call.